core/tests/ts-integration/tests/api/snapshots-creator.test.ts

```diff
@@ -0,0 +1,84 @@
+import { TestMaster } from '../../src/index';
+import fs from 'fs';
+import * as zlib from 'zlib';
+import { snapshots_creator } from 'zk/build/run/run';
+describe('Snapshots API tests', () => {
+    let testMaster: TestMaster;
+
+    beforeAll(() => {
+        testMaster = TestMaster.getInstance(__filename);
+
+        if (process.env.ZKSYNC_ENV!.startsWith('ext-node')) {
+            console.warn("You are trying to run snapshots creator tests on external node. It's not supported.");
+        }
+    });
+
+    async function runCreator() {
+        console.log('Starting creator');
+        await snapshots_creator();
+    }
+
+    async function rpcRequest(name: string, params: any) {
+        const response = await testMaster.mainAccount().provider.send(name, params);
+        console.log(response);
+        return response;
+    }
+
+    async function getAllSnapshots() {
+        return await rpcRequest('snapshots_getAllSnapshots', []);
+    }
+
+    async function getSnapshot(snapshotL1Batch: number) {
+        return rpcRequest('snapshots_getSnapshot', [snapshotL1Batch]);
+    }
+
+    async function decompressGzip(filePath: string): Promise<string> {
+        return new Promise((resolve, reject) => {
+            const readStream = fs.createReadStream(filePath);
+            const gunzip = zlib.createGunzip();
+            let data = '';
+
+            gunzip.on('data', (chunk) => (data += chunk.toString()));
+            gunzip.on('end', () => resolve(data));
+            gunzip.on('error', reject);
+
+            readStream.pipe(gunzip);
+        });
+    }
+    async function createAndValidateSnapshot() {
+        let existingBatchNumbers = (await getAllSnapshots()).snapshotsL1BatchNumbers as number[];
+        await runCreator();
+        let newBatchNumbers = (await getAllSnapshots()).snapshotsL1BatchNumbers as number[];
+        let addedSnapshots = newBatchNumbers.filter((x) => existingBatchNumbers.indexOf(x) === -1);
+        expect(addedSnapshots.length).toEqual(1);
```

Comment:
Dumb question: Does this assume that there's some activity regarding the node (e.g., other tests) run since the last snapshot was created? Otherwise, the node may be at the same snapshot as previously, which would lead to this check failing.

Reply:
Yes, this test implicitly expects that or that it will be run exactly once since server start. I plan to work on this in the future (a solution is to start a fresh zk sync node for such test). In general, unfortunately there are a few such caveats built-in into ts-integration tests